### PR TITLE
Feature: PHP 7.4 Compatibility

### PR DESCRIPTION
**Pull request by reinhardtzundorf**
Tests are passing on my local environment (PHP 7.4.3).

### CODE REVIEW

**Review by reinhardtzundorf — COMMENTED** [review]
Checked code changes.

**Review by reinhardtzundorf — COMMENTED** [review]
Checked all PR changes following review.

File: tests/DTO/DTOTest.php [hunk 1]
```diff
@@ -74,7 +73,7 @@ public function testIntFrom() {
         $exception = false;
         try {
             DTO::intFrom(["v"=>"0x539"], "v");
-        } catch (\Exception) {
+        } catch (\Exception $e) {
```
Comment (flumonion) [inline tests/DTO/DTOTest.php:76]: Let's try and implement exception testing the [PHPUnit](https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#testing-exceptions) way.

Perhaps like this:

```php
$this->expectException(InvalidArgumentException::class);
DTO::intFrom(["v"=>"0x539"], "v");
``` 

See if you spot any other places we are doing this...

**Review by flumonion — CHANGES_REQUESTED** [review]

File: tests/DTO/DTOTest.php (hunk 1 above)
Comment (mozey) [reply tests/DTO/DTOTest.php:76]: I recall having some issues with `expectException()` not working the way it's supposed to a few years ago. I can't remember what the issue was, if it was related to PHPUnit, or the way we setup stock2shop/app. So I've avoided using it in general. Maybe that has been fixed now.

Comment (reinhardtzundorf) [reply tests/DTO/DTOTest.php]: > I recall having some issues with `expectException()` not working the way it's supposed to a few years ago. I can't remember what the issue was, if it was related to PHPUnit, the PHP version, or the way we setup stock2shop/app. So I've avoided using it in general. Maybe that has been fixed now.

This [article](https://thephp.cc/articles/questioning-phpunit-best-practices) by Sebastian Bergmann seems to suggest that it is the best approach.

Comment (reinhardtzundorf) [reply tests/DTO/DTOTest.php]: @flumonion 

Updated the `DTOTest` class, please see method [here](https://github.com/stock2shop/share/blob/ac42b2ec8065220cdffc43c3309480c2e31edad8/tests/DTO/DTOTest.php#L134-L138).

I've replaced all try-catch blocks with a method call:

```php
$this->assertExceptionWithMessage(InvalidArgumentException::class, 'value is non-numeric');
```

**Pull request merged**